client_code/portable/_serialize.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 anvilistas
+
+from ._builtins import registered_builtins
+from ._register import get_registered_cls, registered_types
+
+__version__ = "0.0.1"
+
+VALUE = "_"
+PATHS = "P"
+TYPES = "T"
+UNHANDLED = "X"
+
+
+def serialize_portable_class(obj, cls, path, paths, types, unhandled):
+    __serialize__ = getattr(obj, "__serialize__", None)
+    if __serialize__ is not None:
+        data = __serialize__(None)
+        rv = do_remap(data, path, paths, types, unhandled)
+    else:
+        # we don't need to sort this dict
```

Comment:
I don't get what the comment is trying to say.

Reply:
i'll adjust that - all dicts get serialized as a list of items.
This way we guarantee order across the wire
but in this case we don't care about the order of the `__dict__` so just send it as a dict